network/mode/bridge/bridge.go

```diff
@@ -38,62 +36,95 @@ func New(ctx context.Context, config network.BridgeConfig, manager mgr.NetworkMg
 	}
 
 	// get bridge ip
-	bridgeIP := utils.StringDefault(config.IP, DefaultIPNet)
-	ipNet, err := netlink.ParseIPNet(bridgeIP)
+	bridgeIP := utils.StringDefault(config.IPv4, DefaultIPv4Net)
+	ipv4Net, err := netlink.ParseIPNet(bridgeIP)
 	if err != nil {
 		return fmt.Errorf("failed to parse ip %v", bridgeIP)
 	}
-	logrus.Debugf("initialize bridge network, bridge ip: %s.", ipNet)
+	logrus.Debugf("initialize bridge network, bridge ip: %s.", ipv4Net)
 
 	// init host bridge network.
-	_, err = initBridgeDevice(bridgeName, ipNet)
+	_, err = initBridgeDevice(bridgeName, ipv4Net)
 	if err != nil {
 		return err
 	}
 
 	// get bridge subnet
-	_, subnet, err := net.ParseCIDR(bridgeIP)
+	_, subnetv4, err := net.ParseCIDR(bridgeIP)
 	if err != nil {
 		return fmt.Errorf("failted to parse subnet %v", bridgeIP)
 	}
-	logrus.Debugf("initialize bridge network, bridge network: %s", subnet)
+	logrus.Debugf("initialize bridge network, bridge network: %s", subnetv4)
 
 	// get ip range
-	var ipRange string
-	if config.FixedCIDR != "" {
-		ipRange = config.FixedCIDR
+	var ipv4Range string
+	if config.FixedCIDRv4 != "" {
+		ipv4Range = config.FixedCIDRv4
 	} else {
-		ipRange = subnet.String()
+		ipv4Range = subnetv4.String()
 	}
-	logrus.Debugf("initialize bridge network, bridge ip range in subnet: %s", ipRange)
+	logrus.Debugf("initialize bridge network, bridge ip range in subnet: %s", ipv4Range)
 
 	// get gateway
-	gateway := DefaultGateway
+	gatewayv4 := DefaultGatewayv4
 	if config.GatewayIPv4 != "" {
-		gateway = config.GatewayIPv4
+		gatewayv4 = config.GatewayIPv4
 	}
-	logrus.Debugf("initialize bridge network, gateway: %s", gateway)
+	logrus.Debugf("initialize bridge network, gateway: %s", gatewayv4)
 
 	ipamV4Conf := types.IPAMConfig{
 		AuxAddress: make(map[string]string),
-		Subnet:     subnet.String(),
-		IPRange:    ipRange,
-		Gateway:    gateway,
+		Subnet:     subnetv4.String(),
+		IPRange:    ipv4Range,
+		Gateway:    gatewayv4,
 	}
 
 	ipam := &types.IPAM{
 		Driver: "default",
 		Config: []types.IPAMConfig{ipamV4Conf},
 	}
 
+	if config.IPv6 {
```

Comment:
I think if we support ipv6, we should modify common function, instead of this way:
```
if ipv6 {
    ......
} else {
    ......
}
```
WDYT?

Reply:
Updated according to your comments :-)